### PR TITLE
(PUP-10685) Clean up `ext/` directory

### PR DESCRIPTION
**Pull request by luchihoratiu**
This commit removes files/scripts left over from Ruby Puppet Server (WEBrick and Passenger) that don't seem to be used anymore. 
## Details about remaining contents of `ext/` folder
This directory now contains files used internally when packaging [puppet](https://github.com/puppetlabs/puppet) and [puppet-agent](https://github.com/puppetlabs/puppet-agent)
What follows is a more detailed description of each remaining directory/file:
* `debian/` - init scripts for puppet (used for Debian-based platforms that do not support systemd)
* `hiera/hiera.yaml` - installed to `$codedir/environments/production`as a default Hiera configuration file
* `osx/puppet.plist` - puppet launchd plist for macOS
* `redhat/` -  init scripts for puppet (used for EL-based platforms that do not support systemd)
* `solaris/smf/` - service manifests for Solaris 11
* `suse/client.init` - init script for puppet (used for SUSE-based platforms that do not support systemd) 
* `systemd/puppet.service` - systemd unit file for puppet
* `windows/` - the puppet daemon for Windows, and other useful `.bat` helper wrappers
* `build_defaults.yaml` - information pertaining to the puppetlabs build automation
* `project_data.yaml` - information used when packaging the puppet gem

## TODO:
- [x] Check if [ext/osx/postflight.erb](https://github.com/puppetlabs/puppet/blob/main/ext/osx/postflight.erb) and [ext/osx/preflight.erb](https://github.com/puppetlabs/puppet/blob/main/ext/osx/preflight.erb) are still being used (references found in [puppetlabs/packaging/tasks/apple.rake](https://github.com/puppetlabs/packaging/blob/1.0.x/tasks/apple.rake) but it seems superseded by [puppetlabs/vanagon/lib/vanagon/platform/osx.rb](https://github.com/puppetlabs/vanagon/blob/master/lib/vanagon/platform/osx.rb); not sure)
- [x] Check if [puppetlabs/packaging/tasks/apple.rake](https://github.com/puppetlabs/packaging/blob/1.0.x/tasks/apple.rake) is still being used in pipelines (if not, we should also remove [ext/osx/prototype.plist.erb](https://github.com/luchihoratiu/puppet/blob/PUP-10685/ext/osx/prototype.plist.erb) and [ext/osx/file_mapping.yaml](https://github.com/luchihoratiu/puppet/blob/PUP-10685/ext/osx/file_mapping.yaml))
- [x] Check if [ext/nagios/check_puppet.rb](https://github.com/puppetlabs/puppet/blob/6.x/ext/nagios/check_puppet.rb) is still needed (its defaults are no longer valid but the script can override them)
- [x] Check if [ext/logcheck/puppet](https://github.com/puppetlabs/puppet/blob/6.x/ext/logcheck/puppet) can be removed or it should be updated with puppetserver instead of puppet-master

### CODE REVIEW

File: ext/osx/postflight.erb [hunk 1]
```diff
@@ -1,109 +0,0 @@
-#!/bin/bash
```
Comment (GabrielNagy) [inline ext/osx/postflight.erb:1]: I think these might be used when building dmg files here: https://github.com/puppetlabs/packaging/blob/65185922cbe33f94c16296ba825fe390c4573801/tasks/apple.rake#L61

Comment (GabrielNagy) [reply ext/osx/postflight.erb:1]: But there's also https://github.com/puppetlabs/vanagon/blob/main/lib/vanagon/platform/osx.rb, so I say remove these and see what breaks

Comment (luchihoratiu) [reply ext/osx/postflight.erb:1]: I thought this was also superseded by https://github.com/puppetlabs/vanagon/blob/master/lib/vanagon/platform/osx.rb but seems safer to not remove it 😅 .

**Comment by GabrielNagy** [comment]
I think we can also remove https://github.com/puppetlabs/puppet/blob/d6038363b13cd989d2e7cb35fc9d5c605d31b786/ext/solaris/pkginfo, it seems to refer to version 3.0.0 which was a long time ago.

File: ext/osx/postflight.erb (hunk 1 above)
Comment (luchihoratiu) [reply ext/osx/postflight.erb:1]: Roger Roger

**Comment by GabrielNagy** [comment]
Curious if we still care about https://github.com/luchihoratiu/puppet/blob/PUP-10685/ext/nagios/check_puppet.rb, its defaults are no longer valid but the script can override them.

About https://github.com/luchihoratiu/puppet/blob/PUP-10685/ext/logcheck/puppet, I see it refers to puppet-master instead of puppetserver, should we update it or remove it?

https://github.com/luchihoratiu/puppet/blob/PUP-10685/ext/osx/prototype.plist.erb and https://github.com/luchihoratiu/puppet/blob/PUP-10685/ext/osx/file_mapping.yaml are still used in the apple.rake task from packaging; the question remains if that rake task is used anymore in our pipelines

https://github.com/luchihoratiu/puppet/blob/PUP-10685/ext/suse/puppet.spec this also seems ancient and unmaintained, we can remove it

https://github.com/luchihoratiu/puppet/blob/PUP-10685/ext/suse/server.init this refers to puppetmaster, can be removed

**Comment by luchihoratiu** [comment]
jenkins please test this on all

**Comment by luchihoratiu** [comment]
jenkins please test this on all

**Comment by luchihoratiu** [comment]
jenkins please test this on all

**Comment by luchihoratiu** [comment]
jenkins please test this on all

**Comment by joshcooper** [comment]
The [windows failure](https://github.com/puppetlabs/puppet/pull/8695/checks?check_run_id=3087025157#step:5:337) is unrelated https://tickets.puppetlabs.com/browse/PUP-11122 and can be ignored.

**Comment by joshcooper** [comment]
Looks good, thanks for cleaning this up! I had a few questions/comments:

- [ ] I'm not sure which of the removed files may be used by upstream packaging? For example, [debian maintains a version of puppet with some patches](https://sources.debian.org/src/puppet/6.16.0-1/), mainly to preserve the old confdir, etc paths from Puppet < 4.
- [ ] I think we can remove `ext/README.environments`. It was added in 92dcbb13c1672f0b3ccfff92602a6ad4a7939ef8 and we used to copy into into `/etc/puppet/environments/example_env` for different types of packages. But the creation of the `production` environment is now handled in puppet-agent packaging.
- [ ] It might be nice to add `ext/README` to describe what the remaining files are used for and by whom.
- [ ] I'd be ok removing `ext/logcheck`, `ext/nagios` and `ext/yaml_nodes.rb` and/or moving them to the `examples` directory, so that the only things left are files needed during packaging.

**Comment by luchihoratiu** [comment]
> * [ ]  I'm not sure which of the removed files may be used by upstream packaging? For example, [debian maintains a version of puppet with some patches](https://sources.debian.org/src/puppet/6.16.0-1/), mainly to preserve the old confdir, etc paths from Puppet < 4.
It seems that Debian patches paths in all files (including the `ext` directory). The files shouldn't be used but there will be impact when Debian maintainers might migrate to newer Puppet versions (that include this change).

**Comment by luchihoratiu** [comment]
jenkins please test this on all

**Comment by joshcooper** [comment]
I can't tell if the error is transient or due to removal of some of the osx files?

```
08:21:07 (cd /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.Xb27lMdD/osx/build; /usr/bin/hdiutil create -volname puppet-agent-6.24.0.30.g2e5e0fcc2 -fs JHFS+ -format UDBZ -srcfolder pkg dmg/puppet-agent-6.24.0.30.g2e5e0fcc2-1.osx10.14.dmg)
08:21:11 hdiutil: create failed - Resource busy
08:21:11 make: *** [puppet-agent-6.24.0.30.g2e5e0fcc2-1.osx10.14.dmg] Error 1
```

**Comment by luchihoratiu** [comment]
jenkins please test this on all

**Comment by luchihoratiu** [comment]
jenkins please test this on all

**Comment by luchihoratiu** [comment]
Very weird that these changes shouldn't affect building, especially for specific version of macOS. Even weirder is that step 5 passed in https://github.com/puppetlabs/puppet/pull/8703 where I ran the tests only on macOS 10.14, with this commit cherry-picked but against `main`.

**Comment by luchihoratiu** [comment]
jenkins please test this on all

**Comment by luchihoratiu** [comment]
Acceptance tests passed on all platforms ([here](https://jenkins-platform.delivery.puppetlabs.net/job/platform_puppet-agent-extra_puppet-agent-integration-suite_pr/2231/)) so this should be good for merge ✅ 
Latest push only added a [README](https://github.com/puppetlabs/puppet/blob/d988fe72d98dae26200643e8716a2a93e5235d86/ext/README.md) for the `ext` folder.

**Review by GabrielNagy — APPROVED** [review]

**Pull request merged**